extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -67,6 +68,43 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
     this.storeService_.subscribe(StateProperty.RTL_STATE, (rtlState) => {
       this.onRtlStateUpdate_(rtlState);
     });
+
+    this.storeService_.subscribe(StateProperty.PAGE_SIZE, () => {
+      this.onPageSizeStateUpdate_();
+    });
+  }
+
+  /**
+   * Helper function to check if the shopping tag should flip if it's too far to the right.
+   * @private
+   */
+  shoudTagFlip_() {
+    const {offsetLeft, offsetWidth} = this.element;
+    this.measureElement(() => {
+      /* We only check the right hand side, as resizing only expands the border to the right. */
+
+      const storyPageWidth =
+        this.element.closest('amp-story-page')./*OK*/ offsetWidth;
+      const shouldFlip = offsetLeft + offsetWidth > storyPageWidth;
+
+      this.shoppingTagEl_.classList.toggle(
```

Comment:
You're changing a class in the DOM, which updates the DOM/layout, this needs to be in `mutateElement`

1. Measuring something that might trigger a reflow: measure
2. Writing in the DOM: mutate

Please spend some time looking into the resources I shared with you (or ping me if you want them again / if they're not clear and you want to talk about it). This is really important web perf and frame management stuff.

Reply:
Soudns good, but is it valid to call mutatelement within a measure element?